clients/python/client/osparc/_http_client.py

```diff
@@ -41,3 +44,44 @@ async def __aexit__(self, exc_type, exc_value, traceback) -> None:
                     raise err from exc_value
             await self._client.aclose()
             raise exc_value
+
+    async def _request(
+        self, method: Callable[[Any], Awaitable[httpx.Response]], *args, **kwargs
+    ) -> httpx.Response:
+        @tenacity.retry(
+            reraise=True,
+            wait=tenacity.wait_exponential(
+                multiplier=self.configuration.retries.backoff_factor
+            ),
+            stop=tenacity.stop_after_attempt(
+                self.configuration.retries.total
+                if isinstance(self.configuration.retries.total, int)
+                else 4
+            ),
+            retry=tenacity.retry_if_exception_type(httpx.HTTPStatusError),
+        )
+        async def _():
+            response: httpx.Response = await method(*args, **kwargs)
+            try:
+                response.raise_for_status()
```

Comment:
i think here you can just check with an `if `. Actually this is what they do in the implementaiton of that function. SEE
https://github.com/ITISFoundation/osparc-issues/issues/1136#issuecomment-1943836871

Reply:
Good point. I will correct.

Reply:
Good point, I fixed this